### PR TITLE
add api key docs

### LINKED ISSUE #2: Add description on how to set API key per style
Most the styles must be updated to allow this, though.

**Question:** /cc @bcamper 

* Should Mapzen house styles styles continue to ship with Mapzen's API keys set or leave that blank? 

If we remove it, then our product pages would need to set them on map init, and the Github repo index pages would also need to set them on init.

### PR DESCRIPTION
**Pull request by kkowalsky**
fixes #2 

@nvkelso I put the examples for Tangram JS and the YAML syntax together in the API Reference as it makes more sense. I could add it to 'Get Started' as well, but I don't think it's necessary. 

still unsure if the default = false?

### CODE REVIEW

File: api-reference.md [hunk 1]
```diff
@@ -4,7 +4,39 @@ Mapzen basemaps offer several global properties to customize and extend the map.
...
+
+Using Tangram JS's API:
+```
+Scene.load({
+      import: 'https://mapzen.com/carto/bubble-wrap-style/bubble-wrap.yaml',
+      global: { sdk_api_key: `mapzen-xxxxxx` }
+    });
+```
+
+In the Tangram YAML file:
```
Comment (nvkelso) [inline api-reference.md]: We don't include any other Tangram YAML file snippets for the other API methods. Recommend removing this title and code block.

File: api-reference.md [hunk 2]
```diff
@@ -4,7 +4,39 @@ Mapzen basemaps offer several global properties to customize and extend the map.
 
 Not every basemap supports the full set of resources and the default styling of these assets is customized per Mapzen map style. See [Styles](styles.md) for what's supported.
 
-As the basemaps are still in active development we recommend peggging an import to a specific major version, eg: `5`. See the [versioning](versioning.md) doc for more details.
+As the basemaps are still in active development we recommend pegging an import to a specific major version, eg: `5`. See the [versioning](versioning.md) doc for more details.
+
+## Global API keys
```
Comment (nvkelso) [inline api-reference.md]: Global API keys or Mapzen API keys? Mapzen should either be in the section header or/& also in the paragraph text.

**Comment by nvkelso** [comment]
This is aspirational at the moment since no house style currently supports this API method.

Once they do, we can merge this.

We should probably add a bullet point to each house style in `styles.md` saying that it's not yet supported? (And when it is we'll move that bullet point to the supported column?)

**Review by nvkelso — CHANGES_REQUESTED** [review]

**Comment by nvkelso** [comment]
Yes, the default will be `false` – which means the style would only have access to anonymous rate limits + cached tiles.

File: api-reference.md [hunk 3]
```diff
@@ -4,7 +4,28 @@ Mapzen basemaps offer several global properties to customize and extend the map.
 
 Not every basemap supports the full set of resources and the default styling of these assets is customized per Mapzen map style. See [Styles](styles.md) for what's supported.
 
-As the basemaps are still in active development we recommend peggging an import to a specific major version, eg: `5`. See the [versioning](versioning.md) doc for more details.
+As the basemaps are still in active development we recommend pegging an import to a specific major version, eg: `5`. See the [versioning](versioning.md) doc for more details.
+
+## Mapzen API keys
+
+API keys can be set as a global variable which allows you to easily change them when starting a new project or updating your key. As API keys are not set to a particular product, you can use the same key for using both vector and terrain tile services. This is set in Tangram JS and updates the YAML files as a global variable.
```
Comment (nvkelso) [inline api-reference.md]: This is a generic Tangram feature, not constrained to JS.

**Comment by kkowalsky** [comment]
just noticed that https://mapzen.com/documentation/tangram/android-walkthrough/#using-mapzens-vector-tile-service uses this already in the walkthrough...

File: api-reference.md [hunk 4]
```diff
@@ -4,7 +4,28 @@ Mapzen basemaps offer several global properties to customize and extend the map.
 
 Not every basemap supports the full set of resources and the default styling of these assets is customized per Mapzen map style. See [Styles](styles.md) for what's supported.
 
-As the basemaps are still in active development we recommend peggging an import to a specific major version, eg: `5`. See the [versioning](versioning.md) doc for more details.
+As the basemaps are still in active development we recommend pegging an import to a specific major version, eg: `5`. See the [versioning](versioning.md) doc for more details.
+
+## Mapzen API keys
+
+API keys can be set as a global variable which allows you to easily change them when starting a new project or updating your key. As API keys are not set to a particular product, you can use the same key for using both vector and terrain tile services. This is set in Tangram and updates the YAML files as a global variable.
```
Comment (nvkelso) [inline api-reference.md]: API keys **should** be set using the global variable

File: api-reference.md [hunk 5]
```diff
@@ -4,7 +4,28 @@ Mapzen basemaps offer several global properties to customize and extend the map.
...
+
+### sdk_api_key
+
+* `sdk_api_key`: defaults to `false`
```
Comment (nvkelso) [inline api-reference.md]: defaults to `''`

File: api-reference.md [hunk 6]
```diff
@@ -4,7 +4,28 @@ Mapzen basemaps offer several global properties to customize and extend the map.
...
+
+Example Tangram usage:
+```
+globals:
```
Comment (nvkelso) [inline api-reference.md]: Tangram JS

```
 var layer = Tangram.leafletLayer({
        scene: { import: style_file, global: { sdk_mapzen_api_key: 'mapzen-xxxxxx' } },
        attribution: '<a href="https://mapzen.com/tangram" target="_blank">Tangram</a> | &copy; OSM contributors | <a href="https://mapzen.com/" target="_blank">Mapzen</a>'
    });
```

File: api-reference.md (hunk 4 above)
Comment (nvkelso) [reply api-reference.md]: We should also link to getting an API key...

[Sign up](https://mapzen.com/documentation/overview/#developer-accounts-and-api-keys) for a Mapzen API key to enjoy higher rate limits...

File: api-reference.md (hunk 6 above)
Comment (nvkelso) [reply api-reference.md]: And the Tangram ES version:

```
map->loadScene(
    "https://mapzen.com/carto/bubble-wrap-style/bubble-wrap.yaml",
    false,
    { "global.sdk_mapzen_api_key", "mapzen-xxxxxx" }
);
```

Comment (nvkelso) [reply api-reference.md]: Waiting for Mapzen.js version (should be available next week).

Comment (nvkelso) [reply api-reference.md]: (Remove the snippet of source code, that's internal logic the user shouldn't worry about.)

**Review by nvkelso — APPROVED** [review]

**Pull request merged**